### PR TITLE
project-mongo-api-SandraWallén

### PR DESCRIPTION
**Pull request by sandra-wallen**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -25,11 +65,57 @@ const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Welcome to my mongoDB API!')
```
Comment (TobiasHagberg) [inline server.js:72]: Although this greeting is just fine i quip tip on a package I used is list endpoints from express-list-endpoints which offers en simple guide over avaiable paths in the api

File: server.js [hunk 2]
```diff
@@ -25,11 +65,57 @@ const app = express()
...
+
+app.get('/members/name/:memberName', async (req, res) => {
+  const { memberName } = req.params
+
+  try {
+    const singleMember = await Member.findOne({ name: memberName })
+    res.json(singleMember)
+  } catch (error) {
+    res.status(400).json({ error: 'Something went wrong', details: error })
+  }
```
Comment (TobiasHagberg) [inline server.js:118]: This is probably all on my end but I can't get at member to show using https://sandra-project-mongo-api.herokuapp.com/members/name/poya
I get Null

File: server.js [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/mongoApi"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+const memberSchema = new mongoose.Schema({
+  name: String,
+  surname: String,
+  role: {
+    type: mongoose.Schema.Types.ObjectId,
```
Comment (TobiasHagberg) [inline server.js:19]: This is a bit next level from what I did and I really like the detail of the role as a refernce to another schema: seems useful and probably common in real life to keep objects neet and DBs in check :)

**Review by TobiasHagberg — APPROVED** [review]
Great job Sandra!
This api fulfills all the basic requirements and then some. Sorry to hear about your struggles with deployment and if you get any answers please let me/us know aswell :) 💯

File: server.js (hunk 3 above)
Comment (KarolCislowski) [inline server.js:19]: ❤️

File: server.js [hunk 4]
```diff
@@ -25,11 +65,57 @@ const app = express()
...
+  } catch (error) {
+    res.status(400).json({ error: 'Something went wrong', details: error })
+  }
+})
+
+app.get('/members/name/:memberName', async (req, res) => {
+  const { memberName } = req.params
+
+  try {
+    const singleMember = await Member.findOne({ name: memberName })
```
Comment (KarolCislowski) [inline server.js:114]: you need to add regex here to accept lowercase memberNames

File: server.js [hunk 5]
```diff
@@ -25,11 +65,57 @@ const app = express()
...
+      res.status(400).json({ error: 'Something went wrong', details: error })
+    }
+  } else {
+    try {
+      const allMembers = await Member.find()
+      res.json(allMembers)
+    } catch (error) {
+      res.status(400).json({ error: 'Something went wrong', details: error })
+    }
+  }
```
Comment (KarolCislowski) [inline server.js:96]: nice filtering 👍🏻

**Review by KarolCislowski — APPROVED** [review]
Great API 👍🏻

**Pull request closed without merging**